Framework/ICat/src/CatalogDownloadDataFiles.cpp

```diff
@@ -164,10 +164,8 @@ std::string CatalogDownloadDataFiles::doDownloadandSavetoLocalDrive(
         nullptr, certificateHandler, context);
 
     // Session takes ownership of socket
-    Poco::Net::SecureStreamSocket *socket =
-        new Poco::Net::SecureStreamSocket(context);
-    Poco::Net::HTTPSClientSession session(*socket);
-    socket = nullptr;
+    Poco::Net::SecureStreamSocket socket(context);
```

Comment:
Here be dragons. I think this was allocated on the heap to avoid an error, but when I looked into this a while ago never found where the socket was deleted. Also check that tests for this are enabled (they may have been disabled to avoid network communication in the unit tests).